src/schema/users.ts

```diff
@@ -941,7 +954,9 @@ export const resolvers: IResolvers<any, Context> = {
             )
             .addSelect('u.timezone', 'timezone')
             .innerJoin(User, 'u', `"${builder.alias}"."userId" = u.id`)
-            .where(`"${builder.alias}"."userId" = :id`, { id: ctx.userId }),
+            .where(`"${builder.alias}"."userId" = :id`, {
+              id: id ?? ctx.userId,
+            }),
```

Comment:
Mix and matching between auth and non auth seems complicated, can we maybe just add another userStreakProfile query which will return only stuff needed for profile? It will also avoid below `hasClearedStreak` check.

Reply:
Because even now in tests we are testing for cases where the above can break 😄